client/data/plugins/use-update-schedules-query.ts

```diff
@@ -54,3 +76,69 @@ export const useUpdateScheduleQuery = (
 		...queryOptions,
 	} );
 };
+
+export const useMultisiteUpdateScheduleQuery = (
+	isEligibleForFeature: boolean,
+	queryOptions = {}
+): UseQueryResult< MultisiteSchedulesUpdates[] > => {
+	const state = useSelector( ( state ) => state );
+
+	const retrieveSite = useCallback(
+		( siteId: number ) => {
+			return getSite( state, siteId );
+		},
+		[ state ]
+	);
+
+	return useQuery< MultisiteSchedulesUpdatesResponse, Error, MultisiteSchedulesUpdates[] >( {
+		queryKey: [ 'multisite-schedules-update' ],
+		queryFn: () =>
+			wpcomRequest< MultisiteSchedulesUpdatesResponse >( {
+				path: `/hosting/update-schedules`,
+				apiNamespace: 'wpcom/v2',
+				method: 'GET',
+			} ),
+		enabled: isEligibleForFeature,
+		retry: false,
+		select: ( data ) => {
+			const result: MultisiteSchedulesUpdates[] = [];
+
+			for ( const site_id in data.sites ) {
+				for ( const scheduleId in data.sites[ site_id ] ) {
+					const { timestamp, schedule, args, interval, last_run_timestamp, last_run_status } =
+						data.sites[ site_id ][ scheduleId ];
+
+					const existingSchedule = result.find( ( item ) => item.id === scheduleId );
```

Comment:
Since the schedule id will be the same for the plugins set only, Instead of grouping by the schedule id, should we group for scheduleId+schedule+timestamp, otherwise I might have the same plugins set with two different times and they'll end up in the same group. What do you think?

Reply:
Great thinking! I wrongfully assumed timestamp was in the hash already 🤦‍♂️